src/main/java/pl/javastart/task/contracts/CardContract.java

```diff
@@ -0,0 +1,56 @@
+package pl.javastart.task.contracts;
+
+public class CardContract extends Contract {
+
+    public CardContract(double availableFunds) {
+        this.availableFunds = availableFunds;
+    }
+
+    @Override
+    void sendSms() {
+        if (availableFunds > 0) {
+            availableFunds -= SMS_COST;
+            smsSents += 1;
+            System.out.println("\nSMS Wysłany");
+        } else {
+            System.out.println("\nNie udało się wysłać SMSa");
+        }
+    }
+
+    @Override
+    void call(int seconds) {
+        if (availableFunds > 0) {
+            availableFunds -= ((double) seconds / 60) * CALL_COST_PER_MINUTE;
+            secondsUsed += seconds;
+            System.out.printf("\nPołączenie trwało %d sekund\n", seconds);
```

Comment:
Brakuje zabezpieczenia przed tym, że nie mamy wystarczających środków do wykonania rozmowy. Jeżeli mamy np. 1zł, a rozmowa kosztuje 20gr/min, to po 5 minutach powinno nas rozłączyć.
U Ciebie robi się manko na koncie